# Carl Ritson has contributed this patch as a part of his implementation of RVMThread.setPriority(int).
# This patch ensures that newly created threads will inherit the correct priority even when it is set while they are running.
# As of 2013-08-13 this bug has not been reported upstream at GNU Classpath.
--- java/lang/Thread.java 2013-07-25 14:08:12.000000000 +0100
+++ java/lang/Thread.java 2013-07-25 14:04:48.000000000 +0100
@@ -374,8 +374,8 @@
         this.threadId = ++totalThreadsCreated;
       }
 
-    priority = current.priority;
-    daemon = current.daemon;
+    priority = current.getPriority();
+    daemon = current.isDaemon();
     contextClassLoader = current.contextClassLoader;
     contextClassLoaderIsSystemClassLoader =
         current.contextClassLoaderIsSystemClassLoader;
